### PR TITLE
Add ex_webrtc_endpoint

### PR DESCRIPTION
**Pull request by roznawsk**

### CODE REVIEW

**Review by Karolk99 — CHANGES_REQUESTED** [review]
I think we should remove WebRTC from this PR. We did last release with previous commit and we do not wish to support it further. All changes related to the WebRTC endpoint should be merged into a separate branch dedicated to WebRTC endpoint maintenance.

File: engine/mix.exs [hunk 1]
```diff
@@ -53,15 +53,16 @@ defmodule Membrane.RTC.Engine.MixProject do
 
   defp deps do
     [
-      {:membrane_core, "~> 1.1.2"},
-      {:membrane_rtp_plugin, "~> 0.28.0"},
+      {:membrane_core, "~> 1.0"},
```
[the remark this answers is not in the document]
Comment (roznawsk) [reply engine/mix.exs]: Done

File: ex_webrtc/integration_test/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Integration tests
+
+Directory containing test scenarios for running integration against `membrane_rtc_engine` with the
+use of containerised headless browsers and containerised `test_videoroom` app.
```
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/integration_test/README.md]: Yes, this is the app that we use for integration tests

File: ex_webrtc/internal_docs/assets/ex_webrtc_endpoint_architecture.drawio.svg
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/internal_docs/assets/ex_webrtc_endpoint_architecture.drawio.svg:1]: This is already updated

File: ex_webrtc/internal_docs/protocol.md
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/internal_docs/protocol.md:1]: Yes, the flow didn't change

File: ex_webrtc/lib/metrics.ex
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/lib/metrics.ex:1]: Currently we will have only non-RTP related metrics, such as Metadata updates

File: examples/ex_webrtc_videoroom/README.md
[the remark this answers is not in the document]
Comment (roznawsk) [reply examples/ex_webrtc_videoroom/README.md:1]: Yes it does

File: ex_webrtc/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Membrane RTC Engine ExWebRTC Endpoint
+
+[![CircleCI](https://circleci.com/gh/fishjam-cloud/membrane_rtc_engine.svg?style=svg)](https://circleci.com/gh/fishjam-cloud/membrane_rtc_engine)
+
+ExWebRTC Endpoint for the [Membrane RTC Engine](https://github.com/jellyfish-cloud/membrane_rtc_engine)
```
Comment (Karolk99) [inline ex_webrtc/README.md:5]:
```suggestion
ExWebRTC Endpoint for the [Membrane RTC Engine](https://github.com/fishjam-cloud/membrane_rtc_engine)
```

File: ex_webrtc/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Client SDKs
+
+- [JS/TS](https://github.com/jellyfish-dev/membrane-webrtc-js)
+- [Android](https://github.com/jellyfish-dev/membrane-webrtc-android)
+- [iOS](https://github.com/jellyfish-dev/membrane-webrtc-ios)
+- [React Native](https://github.com/jellyfish-dev/react-native-membrane-webrtc)
```
Comment (Karolk99) [inline ex_webrtc/README.md:12]:
```suggestion
- [JS/TS](https://github.com/fishjam-cloud/membrane-webrtc-js)
- [Android](https://github.com/fishjam-cloud/membrane-webrtc-android)
- [iOS](https://github.com/fishjam-cloud/membrane-webrtc-ios)
- [React Native](https://github.com/fishjam-cloud/react-native-membrane-webrtc)
```

File: ex_webrtc/mix.exs [hunk 5]
```diff
@@ -0,0 +1,153 @@
+defmodule Membrane.RTC.Engine.Endpoint.ExWebRTC.MixProject do
+  use Mix.Project
+
+  @version "0.1.0-dev"
+  @engine_github_url "https://github.com/jellyfish-dev/membrane_rtc_engine"
```
[the remark this answers is not in the document]
Comment (Karolk99) [reply ex_webrtc/mix.exs:5]: .

**Review by Karolk99 — APPROVED** [review]

File: ex_webrtc/integration_test/test_videoroom/README.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# TestVideoroom
+Test application for running integration against `membrane_rtc_engine` - specifically, the WebRTC
```
[the remark this answers is not in the document]
Comment (Karolk99) [reply ex_webrtc/integration_test/test_videoroom/README.md]: .

File: ex_webrtc/lib/metrics.ex
[the remark this answers is not in the document]
Comment (Karolk99) [reply ex_webrtc/lib/metrics.ex:1]: what about `Membrane.RTP.metrics`?

**Review by Karolk99 — COMMENTED** [review]
Generally there is a few requested changes and tests two tests fails

File: ex_webrtc/integration_test/test_videoroom/README.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+To gather current statistics the `playwright` triggers a button in the web page that starts the process of gathering
+the stats. Once the stats are ready, they get put inside the prepared div element and a `data-version` attribute of that div gets incremented.
+
+The version attribute is necessary to differentiate between separate stats button clicks. The counter starts with `0` and for each completed stats
+gathering it gets incremented by `1`. `playwright` can use that information to wait for the div to get updated to this certain counter value
+(it may seem like a hack but it is probably the only way to get around the limitation of the web driver and asynchronous nature of JS code).
+
+So in all tests a version for each client's sessions is maintained and gets incremented once `playwright` extracts the statistics.
+
+## Testing scenarios
```
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/integration_test/test_videoroom/README.md]: The scenarios have been updated (removed simulcast), others are same

File: ex_webrtc/lib/metrics.ex
[the remark this answers is not in the document]
Comment (roznawsk) [reply ex_webrtc/lib/metrics.ex:1]: We don't use `membrane_rtp_plugin` so we don't have any RTP metrics. We should implement them, using the `ExWebRTC.PeerConnection.get_stats()` API

**Comment by roznawsk** [comment]
Updated all occurences of `jellyfish-dev` and changed to `fishjam-cloud`. Added the project to codecov for coverage tracking. Updated badges for codecov and circleci.

File: ex_webrtc/lib/metrics.ex
[the remark this answers is not in the document]
Comment (Karolk99) [reply ex_webrtc/lib/metrics.ex:1]: We should sum up changes in metrics if either fishjam or cloud uses them. This way it'll be easier to adjust our cloud infrastructure to upcoming changes in membrane_rtc_engine.

File: ex_webrtc/README.md (hunk 3 above)
Comment (Karolk99) [reply ex_webrtc/README.md:6]: .

**Review by Karolk99 — APPROVED** [review]

**Pull request merged**